src/documentation/io/built-in-google-bigquery.md

```diff
@@ -0,0 +1,753 @@
+%}```
+
+You can also use BigQuery's standard SQL dialect with a query string, as shown
+in the following example:
+
+```java
+{% github_sample /davidcavazos/beam/blob/bigquery-snippets/examples/java8/src/main/java/org/apache/beam/examples/snippets/Snippets.java tag:BigQueryReadQueryStdSQL
+%}```
+```py
+{% github_sample /davidcavazos/beam/blob/bigquery-snippets/sdks/python/apache_beam/examples/snippets/snippets.py tag:model_bigqueryio_read_query_std_sql
+%}```
+
+
+## Writing to BigQuery
+
+BigQueryIO allows you to write to BigQuery tables. If you are using the Beam SDK
+for Java, you can also write different rows to different tables. BigQueryIO
+writes use APIs that are subject to BigQuery's [Quota](https://cloud.google.com/bigquery/quota-policy#export)
+and [Pricing](https://cloud.google.com/bigquery/pricing) policies.
+
+When you apply a write transform, you must provide the following information:
+
+ * The table name.
+ * The target table's create disposition. The create disposition specifies
+   whether the target table must exist or can be created by the write operation.
+ * The target table's write disposition. The write disposition specifies whether
+   the data you write will replace an existing table, append rows to an existing
+   table, or write only to an empty table.
+
+In addition, if your write operation creates a new BigQuery table, you must also
+supply a table schema for the target table.
+
+
+### Create disposition
+
+The create disposition controls whether or not your BigQuery write operation
+should create a table if the target table does not exist.
+
+<!-- Java specific -->
+
+{:.language-java}
+Use `.withCreateDisposition` to specify the create disposition. Valid enum
+values are:
+
+{:.language-java}
+ * `Write.CreateDisposition.CREATE_IF_NEEDED`: Specifies that the
+   write operation should create a new table if one does not exist. If you use
+   this value, you must provide a table schema with the `withSchema` method.
+   `CREATE_IF_NEEDED` is the default behavior.
+
+ * `Write.CreateDisposition.CREATE_NEVER`: Specifies that a table
+   should never be created. If the target table does not exist, the write
+   operation fails.
+
+<!-- Python specific -->
+
+{:.language-py}
+Use the `create_disposition` parameter to specify the create disposition. Valid
+enum values are:
+
+{:.language-py}
+ * `BigQueryDisposition.CREATE_IF_NEEDED`: Specifies that the write operation
+   should create a new table if one does not exist. If you use this value, you
+   must provide a table schema. `CREATE_IF_NEEDED` is the default behavior.
+
+ * `BigQueryDisposition.CREATE_NEVER`: Specifies that a table should never be
+   created. If the target table does not exist, the write operation fails.
+
+
+If you specify `CREATE_IF_NEEDED` as the create disposition and you don't
+supply a table schema, the transform may fail at runtime with a
+<span class="language-java">java.lang.IllegalArgumentException</span>
```

Comment:
It's not guaranteed that it will be this particular exception. Just say that it may fail.

Reply:
Done